src/storage/v2/storage.cpp

```diff
@@ -185,6 +190,10 @@ Result<std::optional<std::pair<VertexAccessor, std::vector<EdgeAccessor>>>> Stor
     return res.GetError();
   }
 
+  if (flags::run_time::GetExperimentalTextSearchEnabled()) {
+    storage_->indices_.text_index_->RemoveNode(vertex->vertex_);
```

Comment:
You sure you should call this here?

Reply:
This is the point in the code immediately after the vertices have been deleted from Memgraph. Given that, it seemed like the best place for removing those nodes from the text indices as well.